substrate/frame/system/src/lib.rs

```diff
@@ -1474,6 +1481,7 @@ impl<T: Config> Pallet<T> {
 		// keep genesis hash
 		if !to_remove.is_zero() {
 			<BlockHash<T>>::remove(to_remove);
+			<StateRoot<T>>::remove(to_remove);
```

Comment:
It seems `StateRoot` is only stored for the last `BlockHashCount` number of blocks, since it is intended to be used for fraud proof verification this may not be enough because the challenge period of ER is **`BlockTreePruningDepth`** number of **domain blocks**.